src/plugins/intel_gpu/src/graph/assign.cpp

```diff
@@ -43,4 +43,8 @@ void assign_inst::load(cldnn::BinaryInputBuffer& ib) {
     ib >> variable_id;
     set_variable_id(variable_id);
 }
+
+void assign_inst::on_execute() {
+    _outputs[0] = input_memory_ptr(0);
```

Comment:
Only with doing this, don't we have dat arace like 
readvalue -> op (e.g., concat)  -> assign sequence, where assign and readvalue will have same memory, i.e., concat will use same input/ output ? 

To prevent such a problem, I added a new flag to enforce the mem alloc of such concat in the above pattern in this draft https://github.com/openvinotoolkit/openvino/pull/19194/files, shouldnt we need to do such handling ?

Reply:
Self answer) Ah I was confused with the variable. output is okay to set as input.. The "reuse mem of variable with mem of input " will cause the problem I mentioned. And this will be handled by Andrew in the upcoming work